cabal-ghc921.project

```diff
@@ -55,12 +55,10 @@ constraints:
     +ignore-plugins-ghc-bounds
```

Comment:
Shouldn't we be able to go back to just turning this flag off and letting cabal pick the ones which say they can be included based on the ghc bounds in the cabal files?

Reply:
I am afraid we can't, the ghc "bounds" are not a dependency on ghc in `build-depends` but a specific cabal syntax for conditionals using the current ghc, the condition sets the flag and then comes the solver phase.
For example it is possible to have conditions on ghc version without depending at all on ghc as a library

Reply:
Maybe we could replace flags with ghc bounds in `build-depends` (maybe you were thinking on that?) but not all plugins depend on ghc directly but through ghcide and that dependency would be "artificial". I would say plugins should depend less directly on ghc (via hie-compat)

Reply:
I meant:
- We have the `stylish-haskell` flag in `haskell-language-server.cabal`
- We depend on the `stylish-haskell` plugin if the flag is on *and* the GHC version is okay, unless we set the override flag
- Here we're turning off the `stylish-haskell` flag *and* setting the override flag.
- So... what if we stop setting either of them? Then the `stylish-haskell` plugin just won't be enabled because of the GHC version guard, which is what we want, right?

Reply:
hmm, i think the idea was being able to override the ghc bound withouth changing the .cabal file only the cabal.project and change only the .cabal file when the development is finished and we are about to do the hackage release

so you could override it doing a cabal install from a hackage version if you local environment makes the plugin work somewhat

but maybe it is better to update directly the .cabal file yeah as that use case is in fact very rare, will try it, thanks!

Reply:
hmm i think the explanation is we can have plugins working in github but no in hackage (due to allow-newers and remote-source-repo in the cabal.project)
so we have to make them not buildables by default for hackage but skip it in the cabal.project
does it make sense?